docs/src/pages/components/text/demo.tsx

```diff
@@ -1,159 +1,90 @@
-        <div className="w-1/2">
-          <input
-            type="text"
-            id="font_weight"
-            name="font_weight"
-            value={fontWeight}
-            onChange={(event) => setFontWeight(event.target.value)}
-          />
-        </div>
-      </div>
-      <div className="flex mb-4">
-        <div className="w-1/2 text-right pr-2 text-lg mt-1">
-          <label htmlFor="font_size">Font Size:</label>
-        </div>
-        <div className="w-1/2">
-          <input
-            type="text"
-            id="font_size"
-            name="font_size"
-            value={fontSize}
-            onChange={(event) => setFontSize(event.target.value)}
-          />
-        </div>
-      </div>
-      <div className="flex mb-4">
-        <div className="w-1/2 text-right pr-2 text-lg mt-1">
-          <label htmlFor="font_style">Font Style:</label>
-        </div>
-        <div className="w-1/2">
-          <input
-            type="text"
-            id="font_style"
-            name="font_style"
-            value={fontStyle}
-            onChange={(event) => setFontStyle(event.target.value)}
-          />
-        </div>
-      </div>
-      <div className="flex mb-4">
-        <div className="w-1/2 text-right pr-2 text-lg mt-1">
-          <label htmlFor="text_decoration">Text Decoration:</label>
-        </div>
-        <div className="w-1/2">
-          <input
-            type="text"
-            id="text_decoration"
-            name="text_decoration"
-            value={textDecoration}
-            onChange={(event) => setTextDecoration(event.target.value)}
-          />
-        </div>
-      </div>
+import { Demo } from '@/components/Demo';
+import { TextPropControls } from './TextPropControls';
+import { useTextProps } from './useTextPropControlProps';
 
-      <div className="flex mb-4">
-        <div className="w-1/2 text-right pr-2 text-lg mt-1">
-          <label htmlFor="is_truncated">Is Truncated</label>
-        </div>
-        <div className="w-1/2">
-          <input
-            type="checkbox"
-            id="is_truncated"
-            name="is_truncated"
-            onChange={(event) => setIsTruncated(!isTruncated)}
-          />
-        </div>
-      </div>
-      <Example>
-        <Text
-          isTruncated={isTruncated}
-          color={color}
-          lineHeight={lineHeight}
-          fontWeight={fontWeight}
-          fontStyle={fontStyle}
-          fontSize={fontSize}
-          textDecoration={textDecoration}
-        >
-          {text}
-        </Text>
-      </Example>
-    </div>
-  );
-};
+const propsToCode = (
+  variation: TextVariation,
+  {
+    as,
+    isTruncated,
+    color,
+    lineHeight,
+    fontWeight,
+    fontStyle,
+    fontSize,
+    textDecoration,
+    value,
+  }: TextProps
+) => `
+<Text
+    as="${as}"
+    variation="${variation}"
+    isTruncated="${isTruncated}"
+    color="${color}"
+    lineHeight="${lineHeight}"
+    fontWeight="${fontWeight}"
```

Comment:
Can we show `fontWeight` as a `number`, not as a string? (e.g., `fontWeight=${fontWeight}`)

Reply:
Done